src/TicTacToeModel.java

```diff
@@ -0,0 +1,77 @@
+public class TicTacToeModel {
+    public enum Player {X, O, EMPTY}
+    private final Player[][] board = new Player[3][3];
+    private Player currentPlayer;
+
+    public TicTacToeModel() {
+        currentPlayer = Player.X;
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                board[i][j] = Player.EMPTY;
+            }
+        }
+    }
+
+    public void restartGame(){
+        currentPlayer = Player.X;
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                board[i][j] = Player.EMPTY;
+            }
+        }
+    }
+
+    public void checkBounds(int row, int col) {
+        if (row < 0 || row > 2 || col < 0 || col > 2) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public Player getPosition(int row, int col) {
+        checkBounds(row, col);
+        return board[row][col];
+    }
+
+    public Player getCurrentPlayer() {
+        return currentPlayer;
+    }
+
+    public void makeMove(int row, int col) {
+        checkBounds(row, col);
+        if (board[row][col] != Player.EMPTY) {
+            throw new IllegalArgumentException("Spot is already taken.");
+        }
+        board[row][col] = currentPlayer;
+        if (currentPlayer == Player.X) {
+            currentPlayer = Player.O;
+        }
+        else if (currentPlayer == Player.O) {
+            currentPlayer = Player.X;
+        }
+    }
+
+    public Player getWinner() {
+        if (board[0][0] == Player.X && board[0][1] == Player.X && board[0][2] == Player.X ||
+                board[1][0] == Player.X && board[1][1] == Player.X && board[1][2] == Player.X ||
+                board[2][0] == Player.X && board[2][1] == Player.X && board[2][2] == Player.X ||
+                board[0][0] == Player.X && board[1][0] == Player.X && board[2][0] == Player.X ||
+                board[0][1] == Player.X && board[1][1] == Player.X && board[2][1] == Player.X ||
+                board[0][2] == Player.X && board[1][2] == Player.X && board[2][2] == Player.X ||
+                board[0][0] == Player.X && board[1][1] == Player.X && board[2][2] == Player.X ||
+                board[0][2] == Player.X && board[1][1] == Player.X && board[2][0] == Player.X) {
+            return Player.X;
+        }
+
+        if (board[0][0] == Player.O && board[0][1] == Player.O && board[0][2] == Player.O ||
+                board[1][0] == Player.O && board[1][1] == Player.O && board[1][2] == Player.O ||
+                board[2][0] == Player.O && board[2][1] == Player.O && board[2][2] == Player.O ||
+                board[0][0] == Player.O && board[1][0] == Player.O && board[2][0] == Player.O ||
+                board[0][1] == Player.O && board[1][1] == Player.O && board[2][1] == Player.O ||
+                board[0][2] == Player.O && board[1][2] == Player.O && board[2][2] == Player.O ||
```

Comment:
same logic for X and O, so refactor code. DRY

Reply:
Consider using a loop or a helper method to reduce repetition in checking for Player X and Player O. This will make the code more concise and easier to maintain.